app/src/main/java/chat/rocket/android/authentication/login/ui/LoginFragment.kt

```diff
@@ -95,19 +117,160 @@ class LoginFragment : Fragment(), LoginView {
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         if (resultCode == Activity.RESULT_OK) {
-            if (requestCode == REQUEST_CODE_FOR_CAS) {
-                data?.apply {
+            when (requestCode) {
```

Comment:
After `if (resultCode == Activity.RESULT_OK) { ... } ` line you can add `if (data != null) { ... }`. In that way you'll eliminate all the `data!!.something()` and `data?.something()`.